src/Views/OperatingSystemView.vala

```diff
@@ -713,4 +731,116 @@ public class About.OperatingSystemView : Gtk.Box {
+
+            var level_bar = new Gtk.LevelBar ();
+
+            var link_image = new Gtk.Image.from_icon_name ("adw-external-link-symbolic");
+
+            var grid = new Gtk.Grid () {
+                column_spacing = 12,
+                row_spacing = 6,
+                margin_start = 6,
+                margin_end = 6
+            };
+            grid.attach (image, 0, 0, 1, 2);
+            grid.attach (labels_stack, 1, 0, 1, 1);
+            grid.attach (level_bar, 1, 1, 1, 1);
+            grid.attach (link_image, 2, 0, 2, 2);
+
+            child = grid;
+            add_css_class ("link");
+            get_goal_progress (level_bar, target_label);
+
+            Timeout.add (3000, () => {
+                labels_stack.visible_child_name = labels_stack.visible_child_name == "title" ? "goal" : "title";
+                return true;
+            });
+        }
+
+        private void get_goal_progress (Gtk.LevelBar levelbar, Gtk.Label target_label) {
+            new Thread<void*> ("get_goal_progress", () => {
+                var token = About.GITHUB_TOKEN;
+                var query = "{\"query\": \"query { organization(login: \\\"elementary\\\") { sponsorsListing { activeGoal { percentComplete, targetValue } } } }\"}";
+                var curl_command = """
+                curl -X POST https://api.github.com/graphql \
+                -H "Authorization: Bearer %s" \
+                -H "Content-Type: application/json" \
+                -d '%s'
+                """.printf (token, query);
+
+                try {
+                    string standard_output;
+                    string standard_error;
+                    int exit_status;
+
+                    Process.spawn_command_line_sync (
+                        curl_command,
+                        out standard_output,
+                        out standard_error,
+                        out exit_status
+                    );
+
+                    if (exit_status == 0 && standard_output != null) {
+                        var parser = new Json.Parser ();
+                        parser.load_from_data (standard_output);
+
+                        var root = parser.get_root ();
+                        if (root.get_node_type () == Json.NodeType.OBJECT) {
+                            var sponsors_listing = root.get_object ()
+                                .get_object_member ("data")
+                                .get_object_member ("organization")
+                                .get_object_member ("sponsorsListing")
+                                .get_object_member ("activeGoal");
+
+                            int64 percent_complete = sponsors_listing.get_int_member ("percentComplete");
+                            double target_value = sponsors_listing.get_double_member ("targetValue");
+
+                            levelbar.value = percent_complete / 100.0;
+                            target_label.label = _("%s%% towards $%s per month goal")
```

Comment:
There's actually a library we're supposed to use for formatting currency with localization: https://github.com/elementary/switchboard-plug-locale/blob/1c1b281bc56811d456338c7d64d967b0e60353ab/src/Widgets/Preview.vala#L64C13-L64C21

Reply:
I understand, but I don't know this library in detail and I don't know exactly how to use it.